pkg/lib/registry/registry.go

```diff
@@ -157,35 +157,36 @@ func populate(ctx context.Context, loader registry.Load, graphLoader registry.Gr
 				if err != nil {
 					return err
 				}
-				cleanups := make(chan func(), 1)
-				errs := make(chan error, 1)
+				type unpackedImage struct {
+					to      image.Reference
+					from    string
+					cleanup func()
+					err     error
+				}
+				unpacked := make(chan unpackedImage)
 				for bundle := range bundles {
-					if _, ok := overwriteImageMap[img.Bundle.Package]; !ok {
-						overwriteImageMap[img.Bundle.Package] = make(map[image.Reference]string, 0)
-					}
 					// parallelize image pulls
 					go func(bundle registry.BundleKey, img *registry.ImageInput) {
 						if bundle.CsvName != img.Bundle.Name {
 							to, from, cleanup, err := unpackImage(ctx, reg, image.SimpleReference(bundle.BundlePath))
-							if err != nil {
-								errs <- err
-							}
-							cleanups <- cleanup
-							overwriteImageMap[img.Bundle.Package][to] = from
+							unpacked <- unpackedImage{to: to, from: from, cleanup: cleanup, err: err}
 						} else {
-							overwriteImageMap[img.Bundle.Package][to] = from
+							unpacked <- unpackedImage{to: to, from: from, cleanup: func() { return }, err: nil}
+							// safe because this case is only hit once
```

Comment:
it's only hit once per goroutine, but it's still concurrent access to a map

Reply:
what I meant is there is only one go routine that will ever hit this case